src/cmd/ksh93/sh/xec.c

```diff
@@ -3339,9 +3348,16 @@ int sh_funscope_20120720(Shell_t *shp, int argn, char *argv[], int (*fun)(void *
     shp->topscope = (Shscope_t *)prevscope;
     nv_getval(sh_scoped(shp, IFSNOD));
     shp->end_fn = 0;
-    if (nsig) memcpy((char *)&shp->st.trapcom[0], savstak, nsig);
+    if (nsig) {
```

Comment:
You can omit this since the for loop will execute zero times if nsig is zero.

Reply:
Doh! Never mind. Obviously the other two statements need to be predicated on this.